src/views/TabView/TabBarBottom.js

```diff
@@ -158,6 +164,36 @@ class TabBarBottom extends React.PureComponent<Props> {
     return testIDProps;
   };
 
+  _shouldUseHorizontalTabs() {
+    const { routes } = this.props.navigation.state;
+    const { isLandscape, layout, adaptiveTabs, tabStyle } = this.props;
+
+    if (!adaptiveTabs) {
+      return false;
+    }
+
+    let maxTabBarItemWidth = 125;
+    if (tabStyle) {
```

Comment:
This doesn't handle styles created with `StyleSheet.create` or percentage styles. Change this to:

```js
let maxTabBarItemWidth;

const flattenedTabStyle = StyleSheet.flatten(tabStyle);

if (flattenedTabStyle) {
  if (typeof flattenedTabStyle.width === 'number') {
    maxTabBarItemWidth = flattenedTabStyle.width;
  } else if (typeof flattenedTabStyle.width === 'string' && flattenedTabStyle.endsWith('%')) {
    const width = parseFloat(flattenedTabStyle.width);
    if (Number.isFinite(width)) {
      maxTabBarItemWidth = layout.width * (width / 100);
    }
  } else if (typeof flattenedTabStyle.maxWidth === 'number') {
    maxTabBarItemWidth = flattenedTabStyle.maxWidth;
  } else if (typeof flattenedTabStyle.maxWidth === 'string' && flattenedTabStyle.endsWith('%')) {
    const width = parseFloat(flattenedTabStyle.maxWidth);
    if (Number.isFinite(width)) {
      maxTabBarItemWidth = layout.width * (width / 100);
    }
  }
}

if (!maxTabBarItemWidth) {
  maxTabBarItemWidth = 125;
}
```
 Extract it to a function to simplify.